js/msp/msp_helperfunctions.js

```diff
@@ -0,0 +1,1813 @@
+'use strict';
+MSP.listen(MSP_process_data);
```

Comment:
I removed all the parts in msp_protocol that had sort of "ui updates" or the actual call to process_data when data was received. Instead msp_protocol will trigger MSP.notify() when data is received and then all possible listeners will be notified by this.